packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-preview.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  css,
+  SpinLoader,
+  DocumentIcon,
+  spacing,
+  Overline,
+} from '@mongodb-js/compass-components';
+import type { RootState } from '../../../modules';
+import type { Document } from 'mongodb';
+import { DocumentListView } from '@mongodb-js/compass-crud';
+import HadronDocument from 'hadron-document';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  height: '100%',
+});
+
+const previewHeaderStyles = css({
+  paddingTop: spacing[3],
+  paddingBottom: spacing[3],
+  paddingLeft: spacing[3],
+});
+
+const centerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  height: '100%',
+  padding: spacing[3],
+  textAlign: 'center',
+});
+
+const messageStyles = css({ marginTop: spacing[3] });
+
+const documentListStyles = css({
+  overflow: 'scroll',
+});
+
+type PipelinePreviewProps = {
+  isLoading: boolean;
+  previewDocs: Document[] | null;
+};
+
+const PreviewResults = ({
```

Comment:
I think for the empty pipeline (no stages) we should show the input documents, at least this is what I see [in this figma design](https://www.figma.com/file/5dqmdz5h9tqZ4Ypi2VRMHM/Product-Design%3A-AggEx%3A-See-the-whole-pipeline-in-editor-mode?node-id=120%3A26975)

Reply:
there are two design screens for an empty pipeline in the figma link. I'll check with Claudia and if we need to show the documents, i'll create a follow up.